gnucash/report/trep-engine.scm

```diff
@@ -154,6 +156,27 @@ in the Options panel."))
         (cons #\f (G_ "Frozen"))
         (cons #\v (G_ "Voided"))))
 
+;; Define the hashtag-prefix globally so it can be used in the split-sortvalue and renderer
+;; procedures below; it will be set with the actual user-entered string in gnc:trep-renderer
+(define hashtag-prefix "#")
+
+;; Used by the hashtag sortkey to retrieve first matching hashtag from split
+(define (split-hashtag s)
+  (let* ((htp-trimmed (string-trim hashtag-prefix))
+         (htp (if (< 0 (string-length htp-trimmed)) htp-trimmed "#"))
```

Comment:
instea of `(< 0 (string-length str))` prefer `(not (string-null? str))`

Reply:
Thanks Christopher, I knew there had to be a better way. string-null? it is.

John, scheme is funny for sure. Everything is a function or procedure, even <. So < is a procedure that takes 2 arguments and then compares them as in arg1 < arg2, so 0 < (string-length str).  But yes when you read it in a non-scheme way it looks like it's backward...

Reply:
Darn post-fix notation. `string-null?` is definitely clearer for that reason if no other.

BTW, operators are functions in all languages. C++ even names them (e.g. `operator<()`) and lets you write your own.

Reply:
> Darn post-fix notation

I think you mean prefix notation, no? I think postfix notation would be (0 (string-length str) <). Hardly better ;-)

Reply:
Yeah, prefix. Post-fix is Forth and HP calculators. It sort of makes sense when you're operating on an inverted stack (where you push on the bottom instead of the top). Sort of. I can do it on an HP calculator, but I haven't had to program one for almost 50 years and when you're just calculating something the comparisons are in your head, not in the calculator. I hated Forth and never even got a program finished in it.